spec/minesweeper_spec.rb

```diff
@@ -0,0 +1,328 @@
+require('minesweeper')
+
+describe Minesweeper do
+	describe "#initialize" do
+		it 'creates a new board object' do
+			expect(Board).to receive(:new).with(10,9)
+			Minesweeper.new
+		end
+
+		# QUESTION: Why doesn't this work when I remove the let and before statements above?
+		it 'creates a new player object' do
+			expect(Player).to receive(:new)
+			Minesweeper.new
+		end
+	end
+end
+
+describe Board do
+	let(:new_board){Board.new(10,9)}
+	let(:s){new_board.get_square([1,1])}
+
+  describe "#initialize" do
+  	it 'sets @flags_remaining' do
+  		expect(new_board.flags_remaining).to eq(9)
+  	end
+  end
+
+  describe "#create_board" do
+	  it "fills the board with square objects" do
+	    new_board.create_board(10,9)
+	    expect(new_board.board.flatten.all?{ |square| square.is_a?(Square)}).to eq(true)
+	  end
+
+	  it 'fills the board with 100 objects' do
+	  	new_board.create_board(10,0)
+	  	expect(new_board.board.flatten.length).to eq(100)
+	  end
+	end
+
+	describe 'adjacent mines for the board' do
+		it 'surrounding_mines is 0 if there are no mines surrounding the square' do
+			# Create a small board w/ no mines
+			b = Board.new(2,0)
+			expect(b.get_square([1,1]).surrounding_mines).to eq(0)
+		end
+
+		it 'surrounding_mines is 1 w/ 1 mine surrounding the square' do
+			b = Board.new(2,0)
+			b.get_square([1,2]).make_mine  # Set 1 mine
+			b.add_surrounding_mines				 # Need to reset the mine count
+			expect(b.get_square([1,1]).surrounding_mines).to eq(1)
+		end
+
+		it 'surrounding_mines is 2 w/ 2 mine surrounding the square' do
+			b = Board.new(2,0)
+			b.get_square([1,2]).make_mine  # Set 1 mine
+			b.get_square([2,2]).make_mine  # Set 2 mines
+			b.add_surrounding_mines				 # Need to reset the mine count
+			expect(b.get_square([1,1]).surrounding_mines).to eq(2)
+		end
+
+		it 'surrounding_mines is 3 w/ 3 mine surrounding the square' do
+			b = Board.new(2,0)
+			b.get_square([1,2]).make_mine  # Set 1 mine
+			b.get_square([2,2]).make_mine  # Set 2 mines
+			b.get_square([2,1]).make_mine  # Set 3 mines
+			b.add_surrounding_mines				 # Need to reset the mine count
+			expect(b.get_square([1,1]).surrounding_mines).to eq(3)
+		end
+	end
+
+	# These render tests seem to mess with things, 
+	# let's take a look at them later.
+	describe "#render" do
```

Comment:
These checks aren't required as per the specs - would be a real pain to flesh them out fully... :fearful: